### DISCUSSION TITLE
UNO Platform using Hot Restart?

### DISCUSSION THREAD

**Discussion by ward-horsfall** [unoplatform/uno#15857] [Q&A] [answered]
Hi,

I have some queries regarding UNO platform and for targeting the iOS platform.

It looks like using my iPhone and "Hot Restart" this will save me having access to a MacOS machine. I understand when I finally,
publish to the app store I will need one.

Deploy an iOS app using hot restart.
https://learn.microsoft.com/en-us/dotnet/maui/ios/hot-restart?view=net-maui-8.0

Xamarin Hot Restart
https://learn.microsoft.com/en-gb/xamarin/xamarin-forms/deploy-test/hot-restart

My questions are:

1. Will I be able to publish and directly run to my iphone wihout any limitations?
2. Will the app itself be limited to what it can do?
3. Can I then detach my iPhone from my computer (unplug the USB cable) and keep using that app independently of the computer and Visual Studio?
4. Will I be able to use the iOS simulator on Visual Studio 2022?

Thanks

Ward

**Comment by jeromelaban** [comment]
Thanks for your question. 

> Will I be able to publish and directly run to my iphone wihout any limitations? Will the app itself be limited to what it can do?

Yes, you can publish your app to your phone, though there are limitations on what you can do, such as using some APIs that require specific entitlements and configuration from info.plist (e.g. file picking, geo location, anything that requires user permissions). Also, the application is running using the .NET interpreter, so it may not be as fast as if it were built from a mac.

> Can I then detach my iPhone from my computer (unplug the USB cable) and keep using that app independently of the computer and Visual Studio?

Yes it will, though it's only for devices that are registered to your apple developer account.
 
> Will I be able to use the iOS simulator on Visual Studio 2022?

Yes, if you have a physical mac available, or something like [mac in cloud](https://www.macincloud.com/).

**Comment by ward-horsfall** [comment]
Hi jeromelaban,

Great answer thanks.

In regards to the info you info.plist do you have more information on that in general.

I am regards to this. Is the only way people can test the feature like geolocation by using the iOS simulator ? 

Otherwise then have to publish to App Store?

Thanks

Ward

**Reply by jeromelaban** [reply, marked as answer]
Thanks. It's not possible to alter the info.plist at this time, visual studio does not allow for it. For those scenarios, you'll need to have a actual Mac machine to do so.
